Lg/0.5.6 release notes (#71)
dfinity/docs: modules/release-notes/pages/sdk-release-notes.adoc

The {release} release includes the following new features and capabilities:

> - {proglang}, the Candid interface description language, and the Candid UI now support floating point numbers (float64).

Reviewer: ```suggestion
- {proglang}, and the JavaScript library now support floating point numbers (Float).
```

> - When using the Candid UI, you can view return values in the following formats:
> +
>     - using the interface description language syntax
>     - as values displayed in fields using a browser
>     - as JSON fields.

Reviewer: ```suggestion
- When using the Candid UI, you can view return values in text, UI widget and JSON formats.
```

+
You can switch between the formats for representing values by clicking the results bar.
- Internal updates that are not yet user-visible to support future features.

== Breaking change
The {release} release includes the following changes that might require updates to existing programs:

> - The name of the library recognized by the {proglang} compiler has changed from `stdlib` to `base`.

Reviewer: ```suggestion
- The name of the built-in {proglang} library has changed from `stdlib` to `base`.
- All of the module names in the `base` library starts with an uppercase letter.
- module `principalId` changed to `Principal`.
```

+

> If you have programs that rely on modules `base` library, you must change the `import` statement to use the new library name.

Reviewer: ```suggestion
If you have programs that rely on modules `stdlib`, you must change the `import` statement to use the new library name `base`.
```

For example, you might have previously imported the Array module using the following statement:
+
[source,motoko]
----

> import Array "mo:stdlib/Array";

Reviewer: ```suggestion
import Array "mo:stdlib/array";
```

----
+
With this release, you would change the statement like this:
+
[source,motoko]
----
import Array "mo:base/Array";
----

> - The modules in the Motoko base library are case-sensitive.
> With this release, all of the module names have been modified to start with an uppercase letter.
> If you have programs that refer to modules in the `base` library using lowercase names, you must change the program to use the new naming convention.
> +
> For example, if you previously referred to the `array` module, you would change the program to ref to the `Array` modules.

Reviewer: ```suggestion
Special note for Mac users: The Mac file system is case-insensitive by default, so you might be able to build your {proglang} program using a lower case module name, but the same code will not build in Linux. Please always capitalize the module names, e.g. `import Array "mo:base/Array"`, not `import Array "mo:base/array"`.
```
